### ISSUE TITLE
Could not pass ExplosionEvent$Detonate$Impl to FoxGuard

### ISSUE THREAD

**Issue by treyzania** [FoxDenStudio/FoxGuard#102]
I'm not sure what triggered this, but it's probably due to an explosion, obviously.  We're also running Nations which does do some manipulation of explosions depending on nation/zone flags.

```
[03:24:39 ERROR] [Sponge]: Could not pass ExplosionEvent$Detonate$Impl to Plugin{id=foxguard, name=FoxGuard, description=A world protection plugin built for SpongeAPI. Requires FoxCore., url=https://github.com/FoxDenStudio/FoxGuard, authors=[gravityfox], source=mods/foxguard-sponge-0.21.0-405-server.jar}
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_171]
        at java.util.ArrayList.get(ArrayList.java:433) ~[?:1.8.0_171]
        at net.foxdenstudio.sponge.foxguard.plugin.listener.ExplosionListener.handle(ExplosionListener.java:122) ~[ExplosionListener.class:?]
        at net.foxdenstudio.sponge.foxguard.plugin.listener.ExplosionListener.handle(ExplosionListener.java:53) ~[ExplosionListener.class:?]
        at org.spongepowered.common.event.RegisteredListener.handle(RegisteredListener.java:95) ~[RegisteredListener.class:1.12.2-7.1.0-BETA-33]
        at org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:408) [SpongeEventManager.class:1.12.2-7.1.0-BETA-33]
        at org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:435) [SpongeEventManager.class:1.12.2-7.1.0-BETA-33]
        at org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:427) [SpongeEventManager.class:1.12.2-7.1.0-BETA-33]
        at org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:207) [SpongeImpl.class:1.12.2-7.1.0-BETA-33]
        at net.minecraft.world.Explosion.func_77278_a(SourceFile:742) [amp.class:?]
        at net.minecraft.world.WorldServer.func_72885_a(SourceFile:2889) [oo.class:?]
        at net.minecraft.world.WorldServer.triggerInternalExplosion(SourceFile:2475) [oo.class:?]
        at org.spongepowered.common.interfaces.entity.explosive.IMixinExplosive.detonate(IMixinExplosive.java:49) [IMixinExplosive.class:1.12.2-7.1.0-BETA-33]
        at net.minecraft.entity.monster.EntityCreeper.redirect$onExplode$zga003(SourceFile:685) [acs.class:?]
        at net.minecraft.entity.monster.EntityCreeper.func_146077_cc(SourceFile:248) [acs.class:?]
        at net.minecraft.entity.monster.EntityCreeper.func_70071_h_(SourceFile:158) [acs.class:?]
        at org.spongepowered.common.event.tracking.TrackingUtil.tickEntity(TrackingUtil.java:170) [TrackingUtil.class:1.12.2-7.1.0-BETA-33]
        at net.minecraft.world.WorldServer.redirect$onCallEntityUpdate$zlh000(SourceFile:2695) [oo.class:?]
        at net.minecraft.world.World.func_72866_a(SourceFile:1524) [amu.class:?]
        at net.minecraft.world.WorldServer.func_72866_a(SourceFile:714) [oo.class:?]
        at net.minecraft.world.World.func_72870_g(SourceFile:1500) [amu.class:?]
        at net.minecraft.world.World.func_72939_s(SourceFile:4549) [amu.class:?]
        at net.minecraft.world.WorldServer.func_72939_s(SourceFile:2038) [oo.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(SourceFile:1624) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(SourceFile:349) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(SourceFile:560) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(SourceFile:464) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
```

```
Minecraft: 1.12.2
SpongeAPI: 7.1.0-SNAPSHOT-06016f29
SpongeVanilla: 1.12.2-7.1.0-BETA-33
```

**Comment by ghost** [comment]
This is a known issue, it has been fixed locally, and is in the _cleanup/optimization_ branch, there has not yet been a release with this fix.

This was an unexpected issue, in which the event has no affected locations. I will ping this issue when we make a release with the fix.

**Comment by gravityfox** [comment]
This has been fixed since v0.22.0
